src/vs/editor/contrib/stickyScroll/browser/stickyScroll.ts

```diff
@@ -139,7 +139,7 @@ class StickyScrollController extends Disposable implements IEditorContribution {
 		while (outlineElement) {
 			const kind: SymbolKind = outlineElement.symbol.kind;
 			if (kind === SymbolKind.Class || kind === SymbolKind.Constructor || kind === SymbolKind.Function || kind === SymbolKind.Interface || kind === SymbolKind.Method || kind === SymbolKind.Module) {
-				currentStartLine = outlineElement?.symbol.range.startLineNumber as number;
+				currentStartLine = outlineElement?.symbol.selectionRange.startLineNumber as number;
 				currentEndLine = outlineElement?.symbol.range.endLineNumber as number;
```

Comment:
Perhaps the equivalent should also be done on the next line.

Reply:
Thanks for your feedback
But actually selectionRange doesn't include the whole block, only the block name.

![image](https://user-images.githubusercontent.com/6461899/183333021-3d1ab4f1-8cfe-4b7d-97c6-d4201f362738.png)

so it should sticky scroll from selectionRange.startLineNumber to range.endLineNumber

Reply:
Thanks for the explanation.